test/sanity/code-smell/update-bundled.py

```diff
@@ -0,0 +1,131 @@
+    bundled_libs = set()
+    for filename in glob.glob(os.path.join(basedir, 'lib/ansible/compat/*/__init__.py')):
+        if 'compat/tests' in filename:
+            # compat/tests doesn't bundle any code
+            continue
+        bundled_libs.add(filename)
+
+    bundled_libs.add(os.path.join(basedir, 'lib/ansible/module_utils/distro/__init__.py'))
+    bundled_libs.add(os.path.join(basedir, 'lib/ansible/module_utils/six/__init__.py'))
+    bundled_libs.add(os.path.join(basedir, 'lib/ansible/module_utils/compat/ipaddress.py'))
+    # backports.ssl_match_hostname should be moved to its own file
+    bundled_libs.add(os.path.join(basedir, 'lib/ansible/module_utils/urls.py'))
+
+    return bundled_libs
+
+
+def get_files_with_bundled_metadata(paths, basedir):
+    with_metadata = set()
+    for path in paths:
+        if path == 'test/sanity/code-smell/update-bundled.py':
+            continue
+
+        with open(path, 'rb') as f:
+            body = f.read()
+
+        if BUNDLED_RE.search(body):
+            with_metadata.add(os.path.join(basedir, path))
+
+    return with_metadata
+
+
+def get_bundled_metadata(filename):
+    with open(filename, 'r') as module:
+        for line in module:
+            if line.strip().startswith('_BUNDLED_METADATA'):
+                data = line[line.index('{'):].strip()
+                break
+        else:
+            raise ValueError('Unable to check bundled library for update.  Please add'
+                             ' _BUNDLED_METADATA dictionary to the library file with'
+                             ' information on pypi name and bundled version.')
+        metadata = json.loads(data)
+    return metadata
+
+
+def get_latest_applicable_version(pypi_data, constraints=None):
+    latest_version = "0"
+    if 'version_constraints' in metadata:
+        version_specification = packaging.specifiers.SpecifierSet(metadata['version_constraints'])
+        for version in pypi_data['releases']:
+            if version in version_specification:
+                if LooseVersion(version) > LooseVersion(latest_version):
+                    latest_version = version
+    else:
+        latest_version = pypi_data['info']['version']
+
+    return latest_version
+
+
+if __name__ == '__main__':
+    basedir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../..'))
+
+    bundled_libs = get_bundled_libs(basedir)
+    paths = sys.argv[1:] or sys.stdin.read().splitlines()
+    files_with_bundled_metadata = get_files_with_bundled_metadata(paths, basedir)
+
+    for filename in files_with_bundled_metadata.difference(bundled_libs):
+        print('{0}: ERROR: File contains _BUNDLED_METADATA but needs to be added to'
+              ' test/sanity/code-smell/update-bundled.py'.format(filename))
+
+    for filename in bundled_libs:
+        filename = os.path.normpath(filename)
+        try:
+            metadata = get_bundled_metadata(filename)
+        except ValueError as e:
```

Comment:
Include an exception handler for files that do not exist.

Reply:
Does this need to work on both Python2 and Python3?

Reply:
Used ```except (IOError, OSError)``` so that it will catch File not found on both Python2 and Python3.